tests/integration/targets/eos_config/tests/cli/diff_against_config.yaml

```diff
@@ -0,0 +1,50 @@
+---
+- debug:
+    msg: START cli/diff_against_config.yaml on connection={{ ansible_connection }}
+
+- name: setup
+  become: true
+  arista.eos.eos_config:
+    lines: hostname {{ inventory_hostname_short }}
+    match: none
+
+- block:
+
+    - name: get current running-config
+      become: true
+      register: config_before
+      arista.eos.eos_command:
+        commands: show running-config
+
+    - name: configure hostname
+      become: true
+      register: result
+      diff: true
+      arista.eos.eos_config:
+        intended_config: "hostname foo"
+        diff_against: validate_config
```

Comment:
Just playing around with this and looking at the test case, I think this function is useful.

But I do not think its the ask/vision.  I would want to replicate the replace_config approach, but instead of applying the config at the end of the config_session, it would pull the output of `show session-config`, then abort the config session so nothing is applied.  @carlbuchmann & @titom73 as I do not think I am articulating this well.